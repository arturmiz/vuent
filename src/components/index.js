import './Vnt.vue';
export { default as VntAutosuggest } from './autosuggest';
export { default as VntButton } from './button';
export { default as VntCheckbox } from './checkbox';
export { default as VntDialog } from './dialog';
export { default as VntHeader } from './header';
export { default as VntIcon } from './icon';
export { default as VntInput } from './input';
export { default as VntLink } from './link';
export { default as VntListView } from './listview';
export { default as VntNavView } from './navview';
export { default as VntPivot } from './pivot';
export { default as VntRadio } from './radio';
export { default as VntRating } from './rating';
export { default as VntSelect } from './select';
export { default as VntSlider } from './slider';
export { default as VntToggle } from './toggle';
