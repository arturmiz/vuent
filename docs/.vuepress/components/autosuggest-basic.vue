<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestBasic',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
