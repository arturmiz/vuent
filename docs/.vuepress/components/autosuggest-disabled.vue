<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries" disabled></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestDisabled',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
