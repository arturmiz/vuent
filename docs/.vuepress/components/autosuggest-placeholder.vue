<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"
                     placeholder="Type something..."></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestPlaceholder',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
