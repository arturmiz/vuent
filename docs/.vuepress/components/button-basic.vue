<template>
  <div class="playground">
    <vnt-button>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonBasic',
  components: {
    VntButton
  }
}
</script>
