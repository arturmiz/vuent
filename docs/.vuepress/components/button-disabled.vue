<template>
  <div class="playground">
    <vnt-button disabled>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonDisabled',
  components: {
    VntButton
  }
}
</script>
