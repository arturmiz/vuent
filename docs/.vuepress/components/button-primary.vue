<template>
  <div class="playground">
    <vnt-button primary>Primary</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonPrimary',
  components: {
    VntButton
  }
}
</script>
