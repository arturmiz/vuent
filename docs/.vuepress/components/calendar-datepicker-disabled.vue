<template>
  <div class="playground">
    <vnt-calendar-datepicker v-model="date" disabled />
    <div class="result">date: {{ date }}</div>
  </div>
</template>

<script>
import { VntCalendarDatepicker } from "../../../src/components";

export default {
  name: "CalendarDatepickerBasic",
  components: {
    VntCalendarDatepicker
  },
  data() {
    return {
      date: "2020-04-05"
    };
  }
};
</script>
