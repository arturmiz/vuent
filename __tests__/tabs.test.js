import { createLocalVue, mount } from '@vue/test-utils';
import { VntTabs } from '@/components';
import { isInstalled } from './utils';

describe('Tabs', () => {
  let wrapper;

  test('can be installed separately', () => {
    const localVue = createLocalVue();
    localVue.use(VntTabs);

    expect(isInstalled(localVue, VntTabs)).toBe(true);
  });

  describe('by default', () => {

    beforeAll(() => {
      wrapper = mount(VntTabs);
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('generates correct stucture', () => {

    beforeAll(() => {
      wrapper = mount(VntTabs, {
        slots: {
          default: `
            <vnt-tab label="Pivot Item 1" active>
              <div>Tab 1</div>
            </vnt-tab>
            <vnt-tab label="Pivot Item 2">
              <p>Tab 2</p>
            </vnt-tab>
            <vnt-tab label="Tab 3">
              <p>Tab 3</p>
            </vnt-tab>`
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('once a tab is clicked', () => {

    beforeAll(() => {
      wrapper = mount(VntTabs, {
        slots: {
          default: `
            <vnt-tab label="Pivot Item 1" active>
              <p>Tab 1</p>
            </vnt-tab>
            <vnt-tab label="Pivot Item 2">
              <p>Tab 2</p>
            </vnt-tab>`
        }
      });

      const label = wrapper.findAll('.vnt-tabs__header').at(1);
      label.trigger('click');
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

});
