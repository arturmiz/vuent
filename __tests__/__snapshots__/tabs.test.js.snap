// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs by default renders correctly 1`] = `
<div class="vnt-tabs">
  <ul class="vnt-tabs__navigation"></ul>
  <div class="vnt-tabs__container"></div>
</div>
`;

exports[`Tabs generates correct stucture renders correctly 1`] = `
<div class="vnt-tabs">
  <ul class="vnt-tabs__navigation">
    <li class="vnt-tabs__header vnt-tabs__header--active">Pivot Item 1</li>
    <li class="vnt-tabs__header">Pivot Item 2</li>
    <li class="vnt-tabs__header">Tab 3</li>
  </ul>
  <div class="vnt-tabs__container">
    <div class="vnt-tabs__tab vnt-tabs__tab--active">
      <div>Tab 1</div>
    </div>
    <div class="vnt-tabs__tab">
      <p>Tab 2</p>
    </div>
    <div class="vnt-tabs__tab">
      <p>Tab 3</p>
    </div>
  </div>
</div>
`;

exports[`Tabs once a tab is clicked renders correctly 1`] = `
<div class="vnt-tabs">
  <ul class="vnt-tabs__navigation">
    <li class="vnt-tabs__header">Pivot Item 1</li>
    <li class="vnt-tabs__header vnt-tabs__header--active">Pivot Item 2</li>
  </ul>
  <div class="vnt-tabs__container">
    <div class="vnt-tabs__tab">
      <p>Tab 1</p>
    </div>
    <div class="vnt-tabs__tab vnt-tabs__tab--active">
      <p>Tab 2</p>
    </div>
  </div>
</div>
`;
